### PR TITLE
Rename binaries to strip '-alpha'

### PR DESCRIPTION
**Pull request by petemounce**
##### Checklist

- [x] `make test-all` (UNIX) passes. CI will also test this
- [x] unit and/or integration tests are included (if applicable)
- [x] documentation is changed or added (if applicable)

### Description of change

As discussed in https://github.com/aelsabbahy/goss/pull/663#issuecomment-743127235; strip the -alpha naming, keep the messaging, keep the --use-alpha flag and env-var.

We're reasonably comfortable with the current situation - nothing seems to be horrendously broken in terms of the ported functionality as exercised by the added macOS & Windows integration tests, and based on Improbable's use (that I am intimately familiar with).

This change allows people to remove edge-cases from their installation, yet retains the clearly-set expectations of support being community-driven.

Fixes #651. Continues #663.

**Note:** this is aimed at `v4`.

### CODE REVIEW

File: cmd/goss/goss.go [hunk 1]
```diff
@@ -408,14 +408,14 @@ func addAlphaFlagIfNeeded(app *cli.App) {
...
-			Usage:  fmt.Sprintf("goss is alpha-quality. Set to 1 to use anyway."),
+			Usage:  fmt.Sprintf("goss on macOS/Windows is alpha-quality. Set to 1 to use anyway."),
 			EnvVar: "GOSS_USE_ALPHA",
 			Value:  "0",
 		})
 	}
 }
 
-const msgFormat string = `WARNING: goss for this platform (%q) is alpha-quality, work-in-progress, and not yet exercised within continuous integration.
+const msgFormat string = `WARNING: goss for this platform (%q) is alpha-quality, work-in-progress and community-supported.
```
Comment (petemounce) [inline cmd/goss/goss.go:418]: I figured that the integration tests running in CI merited this rewording.

File: cmd/goss/goss.go [hunk 2]
```diff
@@ -408,14 +408,14 @@ func addAlphaFlagIfNeeded(app *cli.App) {
 	if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
 		app.Flags = append(app.Flags, cli.StringFlag{
 			Name:   "use-alpha",
-			Usage:  fmt.Sprintf("goss is alpha-quality. Set to 1 to use anyway."),
+			Usage:  fmt.Sprintf("goss on macOS/Windows is alpha-quality. Set to 1 to use anyway."),
```
Comment (petemounce) [inline cmd/goss/goss.go:411]: Even though this message should never display on other platforms, I figured this more clearly set expectations.

File: Makefile [hunk 3]
```diff
@@ -36,15 +36,13 @@ bench:
 	$(info INFO: Starting build $@)
 	go test -bench=.
 
-alpha-test-%: release/goss-%
+test-int-validate-%: release/goss-%
```
Comment (petemounce) [inline Makefile:39]: I could equally have called this `test-int-all-commands-%` and similarly `run-command-tests.sh` but I thought that might have been confusing with the `command` assertion specifically; what do you think, @aelsabbahy?

**Comment by petemounce** [comment]
@aelsabbahy I think this is ready to review; I'm just waiting for linux CI to pass. I expect it to, but then, I always expect it to... ;-)

File: Makefile (hunk 3 above)
Comment (aelsabbahy) [reply Makefile:39]: no opinion on this currently.

I have some plans to re-write all of the integration tests.. but want to wait until after `v4`

**Comment by aelsabbahy** [comment]
Awesome timing on this.. I'm really really hoping to have v4 released soon. Doing last-minute merges of breaking features and of-course there's going to be a massive rebase of the transform branch.. hopefully, it's good to go after that!

**Comment by aelsabbahy** [comment]
Quick eyeball check on this looks good.. plus CI passed.

Any idea why codeclimate didn't trigger? I assume none of your changes messed with that.

I'll wait for your answer on codeclimate and merge.. If you didn't change anything, don't worry about why it's not working.. just let me know and I'll merge anyways. =)

**Comment by petemounce** [comment]
> Quick eyeball check on this looks good.. plus CI passed.
> 
> Any idea why codeclimate didn't trigger? I assume none of your changes messed with that.
> 
> I'll wait for your answer on codeclimate and merge.. If you didn't change anything, don't worry about why it's not working.. just let me know and I'll merge anyways. =)

The only thing that springs to mind about codeclimate is maybe it's only configured to trigger against branches aimed at master, or only against branches aimed at protected branches?

**Comment by aelsabbahy** [comment]
> The only thing that springs to mind about codeclimate is maybe it's only configured to trigger against branches aimed at master, or only against branches aimed at protected branches?

Yup, that's my guess too.. just wanted to check you didn't knowingly change anything to impact this.

Related note, I may get rid of codeclimate, or at least run it against the official Go codebase and see how many findings come up. If Official Go codebase fails, then I'm not interested in codeclimate. Also, need to actually have golint fail the build and fix all those findings.

Thanks for the PR, love the cleanups!
